src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -713,15 +716,16 @@ internal override bool Equals(TypeSymbol t2, bool ignoreCustomModifiersAndArrayS
 
             // The checks above are supposed to handle the vast majority of cases.
             // More complicated cases are handled in a special helper to make the common case scenario simple/fast
```

Comment:
You're right. That comment makes no sense to me either.
The comment has been there since the Roslyn hello world: https://github.com/dotnet/roslyn/commit/3611ed35610793e814c8aa25715aa582ec08a8b6

Removed.